private/bufpkg/bufmodule/bufmodule.go

```diff
@@ -446,28 +446,28 @@ func ModuleDigestB3(ctx context.Context, module Module) (string, error) {
 func ModuleToBucket(
 	ctx context.Context,
 	module Module,
-	writeBucket storage.WriteBucket,
+	readWriteBucket storage.ReadWriteBucket,
```

Comment:
Why does this need a `ReadWriteBucket`? `ModuleToBucket` should only need to write

Reply:
This is because `PutDependencyModulePinsToBucket` calls the new `ValidateModulePinsConsistentDigests` to read the current `buf.lock` file, if present, before overwriting it.

Reply:
This feels a bit off IMO - the purpose of this function is to write a module to a bucket, but it seems like it has hidden behavior that says "if that bucket already has a buf.lock, do something" - that would be surprising to me as a caller of this function. What is it verifying, is it saying "if the `module` variable does not match a pre-existing file from `buflock.ReadConfig` then error"? This feels like a separate function that should be called outside of `ModuleToBucket`

Reply:
Do you have a suggestion on where this would be good to add? This is a bit of an edge case but we do want to intercept before writing a buf.lock file so we don't overwrite an existing digest for a given commit with a new one, which could be a security concern.

Reply:
It feels like you should do the validate call before calling ModuleToBucket, wherever you are calling ModuleToBucket. The only non-test place this is called is in module_cacher.go, so this should be easy

Reply:
Updated to reduce the scope of the change to only the `buf mod update` command.